File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/CounterpartDB/CounterpartForm.ts
import { ServiceLookupEditor, EnumEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { CopyType } from "../Modules/Enums.Letter.CopyType";

export interface CounterpartForm {
    ReciverId: ServiceLookupEditor;
    CopyType: EnumEditor;
}

export class CounterpartForm extends PrefixedContext {
    static readonly formKey = 'CounterpartDB.Counterpart';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CounterpartForm.init)  {
            CounterpartForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = EnumEditor;

            initFormType(CounterpartForm, [
                'ReciverId', w0,
                'CopyType', w1
            ]);
        }
    }
}

[CopyType]; // referenced types

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/CounterpartDB/Counterpart/CounterpartForm.cs
using CorrespondenceSystem.Modules.Enums.Letter;
using Serenity.ComponentModel;
using System;

namespace CorrespondenceSystem.CounterpartDB.Forms;

[FormScript("CounterpartDB.Counterpart")]
[BasedOnRow(typeof(CounterpartRow), CheckNames = true)]
public class CounterpartForm
{

    public Guid ReciverId { get; set; }

    public CopyType CopyType { get; set; }

}